carbon-p2-plugin/src/main/java/org/wso2/maven/p2/utils/FileManagementUtil.java

```diff
@@ -0,0 +1,286 @@
+     * @param configIniFile File object representing the config.ini
+     * @param propKey       property key
+     * @param value         property value
+     */
+    public static void changeConfigIniProperty(File configIniFile, String propKey, String value) {
+        Properties prop = new Properties();
+
+        try (InputStream inputStream = new FileInputStream(configIniFile)){
+            prop.load(inputStream);
+            prop.setProperty(propKey, value);
+            try(OutputStream outputStream = new FileOutputStream(configIniFile)) {
+                prop.store(outputStream, null);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Zip a give folder to a give output zip file.
+     *
+     * @param srcFolder   source folder
+     * @param destZipFile path to the output zip file
+     */
+    public static void zipFolder(String srcFolder, String destZipFile) {
+        try(FileOutputStream fileWriter = new FileOutputStream(destZipFile);
+            ZipOutputStream zip = new ZipOutputStream(fileWriter)) {
+            addFolderContentsToZip(srcFolder, zip);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+
+    }
+
+    private static void addToZip(String path, String srcFile, ZipOutputStream zip) {
+        File folder = new File(srcFile);
+        if (folder.isDirectory()) {
+            addFolderToZip(path, srcFile, zip);
+        } else {
+            // Transfer bytes from in to out
+            byte[] buf = new byte[1024];
+            int len;
+            try (FileInputStream inputStream = new FileInputStream(srcFile)){
+
+                if (path.trim().equals("")) {
+                    zip.putNextEntry(new ZipEntry(folder.getName()));
+                } else {
+                    zip.putNextEntry(new ZipEntry(path + "/" + folder.getName()));
+                }
+                while ((len = inputStream.read(buf)) > 0) {
+                    zip.write(buf, 0, len);
+                }
+                inputStream.close();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    private static void addFolderContentsToZip(String srcFolder, ZipOutputStream zip) {
+        File folder = new File(srcFolder);
+        String fileList[] = folder.list();
+        if (fileList != null) {
+            try {
+                int i = 0;
+                while (true) {
+                    if (fileList.length == i) {
+                        break;
+                    }
+                    if (new File(folder, fileList[i]).isDirectory()) {
+                        zip.putNextEntry(new ZipEntry(fileList[i] + "/"));
+                        zip.closeEntry();
+                    }
+                    addToZip("", srcFolder + "/" + fileList[i], zip);
+                    i++;
+                }
+            } catch (IOException e) {
+                e.printStackTrace();
```

Comment:
This is the migrated code. It is refactored in https://github.com/wso2/carbon-maven-plugins/pull/8